packages/modules/index/integration-tests/__tests__/query-builder.spec.ts

```diff
@@ -937,4 +963,108 @@ describe("IndexModuleService query", function () {
       },
     ])
   })
+
+  it("should query products filtering product not in [X]", async () => {
+    const expected = [
+      {
+        id: "prod_2",
+        title: "Product 2 title",
+        deep: {
+          a: 1,
+          obj: {
+            b: 15,
+          },
+        },
+      },
+    ]
+
+    const { data } = await module.query({
+      fields: ["product.*"],
+      filters: {
+        $not: [
+          {
+            product: {
+              id: {
+                $in: ["prod_1"],
+              },
+            },
+          },
+        ],
+      },
+    })
+
+    const { data: dataNot } = await module.query({
+      fields: ["product.*"],
+      filters: {
+        product: {
+          id: {
+            $nin: ["prod_1"],
+          },
+        },
+      },
+    })
+    expect(dataNot).toEqual(expected)
+    expect(data).toEqual(dataNot)
+  })
+
+  it("should query products filtering product not in [X]", async () => {
+    const expected = [
+      {
+        id: "prod_2",
+        title: "Product 2 title",
+        deep: {
+          a: 1,
+          obj: {
+            b: 15,
+          },
+        },
+      },
+    ]
+
+    const { data } = await module.query({
+      fields: ["product.*", "variants.*"],
+      filters: {
+        product: {
+          $and: [
+            {
+              variants: {
+                sku: {
+                  $nin: ["sku 123"],
+                },
+              },
+            },
+            {
+              title: {
+                $eq: "Product 2 title",
+              },
+            },
+          ],
+        },
+      },
+    })
+    expect(data).toEqual(expected)
+
+    const { data: data2 } = await module.query({
```

Comment:
suggestion: this test is supposed to test not in, though we are also testing and, not etc. Should we split these tests a bit differently wdy?